src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -708,19 +708,33 @@ class ColumnReferenceSegment(ObjectReferenceSegment):  # type: ignore
     )
 
     def extract_possible_references(self, level):
-        """Extract possible references of a given level."""
+        """Extract possible references of a given level.
+
+        Overrides the parent-class function. BigQuery's support for things like
+        the following
+        - Functions that take a table as a parameter (e.g. TO_JSON_STRING)
+          https://cloud.google.com/bigquery/docs/reference/standard-sql/
+          json_functions#to_json_string
+        - STRUCT
+
+        mean that, without schema information (which SQLFluff does not have),
+        references to data are often ambiguous.
+        """
         level = self._level_to_int(level)
         refs = list(self.iter_raw_references())
         if level == self.ObjectReferenceLevel.SCHEMA.value and len(refs) >= 3:
             return [refs[0]]  # pragma: no cover
-        if level == self.ObjectReferenceLevel.TABLE.value and len(refs) >= 3:
-            # Ambiguous case: The table could be the first or second part, so
-            # return both.
-            return [refs[0], refs[1]]
-        if level == self.ObjectReferenceLevel.OBJECT.value and len(refs) >= 3:
+        if level == self.ObjectReferenceLevel.TABLE.value:
+            # One part: Could be a table, e.g. TO_JSON_STRING(t)
+            # Two parts: Could be schema+table or table+column.
+            # Three parts: Could be table+column+struct or schema+table+column.
+            return refs[:2]
+        if (
+            level == self.ObjectReferenceLevel.OBJECT.value and len(refs) >= 3
+        ):  # pragma: no cover
```

Comment:
Had to move the "no cover" because, by removing the ` and len(refs) >= 3` above, no we never hit this `if` statement.

Reply:
I don’t get this. By removing part of the if statement how can we make it LESS likely the if statement is hit?

Reply:
Because making the `if` statement less stringent, we are more likely to enter the `if` block, which returns from the function, thus the rest of the function is skipped.

Reply:
In other words, previously the function only exited early for TABLE and a reference length 3 or greater. Now it exits early for **any** table reference.

Reply:
OK but moving the no cover from here:

```py
if if_test1 and test2:
   if_clause # pragma: no cover
other_statement
```

to this:


```py
if (
  if_test1
):  # pragma: no cover
   if_clause
other_statement
```

Doesn't really have any effect. The problem was the `other_statement` was no longer being hit, so that's where the pragma needed to be added. Which you did now in https://github.com/sqlfluff/sqlfluff/pull/2278/commits/90dd808d82cd3518df47dc3428b13f8d1f9fb697

So good to approve now. You could revert this pragma change now before merging if you wanted to remove the need for the bracket and extra lines (which I presume were only added to allow the pragma to fit in?).